src/libmeasurement_kit/ooni/bouncer_impl.hpp

```diff
@@ -0,0 +1,116 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/ext.hpp>
+#include <measurement_kit/http.hpp>
+#include <measurement_kit/ooni.hpp>
+
+using json = nlohmann::json;
+
+namespace mk {
+namespace ooni {
+namespace bouncer {
+
+struct Reply {
+    std::string https_collector;
+    std::string https_helper;
+
+    std::string cloudfront_collector;
+    std::string cloudfront_helper;
+};
+
+template <MK_MOCK_NAMESPACE(http, request)>
+void post_net_tests_impl(std::string base_bouncer_url, std::string test_name,
+                    std::string test_version, std::list<std::string> helpers,
+                    Callback<Error, Var<Reply>> cb, Settings settings,
+                    Var<Reactor> reactor, Var<Logger> logger) {
+
+    json request;
+    json net_tests;
+    net_tests["input-hashes"] = {};
+    net_tests["name"] = test_name;
+    net_tests["test-helpers"] = helpers;
+    net_tests["version"] = test_version;
+    request["net-tests"][0] = net_tests;
+
+    settings["http/url"] = base_bouncer_url;
+    settings["http/method"] = "POST";
+    http_request(
+        settings, {}, request.dump(),
+        [=](Error e, Var<http::Response> resp) {
+            if (e) {
+                reactor->call_soon([=]() { cb(e, nullptr); });
```

Comment:
Here I think we don't need `call_soon()` (or I don't understand why we're doing it)